docs/outlook/smart-alerts-onmessagesend-walkthrough.md

```diff
@@ -258,10 +259,63 @@ In this scenario, you'll add handling for sending a message. Your add-in will ch
 
 1. Add an attachment then send the message again. There should be no alert this time.
 
+## Smart Alerts feature behavior and scenarios
+
+Descriptions of the **SendMode** options and recommendations for when to use them are detailed in [Available SendMode options](/javascript/api/manifest/launchevent). The following describes the feature's behavior for certain scenarios.
+
+### Add-in is unavailable
+
+If the add-in is unavailable when a message or appointment is being sent (for example, an error occurs that prevents the add-in from loading), the user is alerted. The options available to the user differ depending on the **SendMode** option applied to the add-in.
+
+If the `PromptUser` or `SoftBlock` option is used, the user can choose **Send Anyway** to send the item without the add-in checking it, or **Try Later** to let the item be checked by the add-in when it becomes available again.
+
+![Dialog that alerts the user that the add-in is unavailable and gives the user the option to send the item now or later.](../images/outlook-soft-block-promptUser-unavailable.png)
+
+If the `Block` option is used, the user can't send the item until the add-in becomes available.
+
+![Dialog that alerts the user that the add-in is unavailable. The user can only send the item when the add-in is available again.](../images/outlook-hard-block-unavailable.png)
+
+### Long-running add-in operations
+
+If the add-in runs for more than five seconds (but less than five minutes), the user is alerted that the add-in is taking longer than expected to process the message or appointment.
+
+If the `PromptUser` option is used, the user can choose **Send Anyway** to send the item without the add-in completing its check. Alternatively, the user can select **Don't Send** to stop the add-in from processing.
+
+![Dialog that alerts the user that the add-in is taking longer to process the item. The user can choose to send the item without the add-in completing its check or stop the add-in from processing the item.](../images/outlook-promptUser-long-running.png)
+
+However, if the `SoftBlock` or `Block` option is used, the user will not be able to send the item until the add-in completes processing it.
+
+![Dialog that alerts the user that the add-in is taking longer to process the item. The user must wait until the add-in completes processing the item before it can be sent.](../images/outlook-soft-hard-block-long-running.png)
+
+`OnMessageSend` and `OnAppointmentSend` add-ins should be short-running and lightweight. To avoid the long-running operation dialog, use other events to pre-process conditional checks before the `OnMessageSend` or `OnAppointmentSend` event is activated. For example, if the user is required to encrypt attachments for every message or appointment, consider using the `OnMessageAttachmentsChanged` or `OnAppointmentAttachmentsChanged` event to perform the check.
```

Comment:
I don't know that we should say "pre" (of pre-process) and "before" since they mean essentially the same
```suggestion
`OnMessageSend` and `OnAppointmentSend` add-ins should be short-running and lightweight. To avoid the long-running operation dialog, use other events to process conditional checks before the `OnMessageSend` or `OnAppointmentSend` event is activated. For example, if the user is required to encrypt attachments for every message or appointment, consider using the `OnMessageAttachmentsChanged` or `OnAppointmentAttachmentsChanged` event to perform the check.
```